nipype/interfaces/mrtrix3/preprocess.py

```diff
@@ -23,20 +23,27 @@ class DWIDenoiseInputSpec(MRTrix3BaseInputSpec):
         argstr='-mask %s',
         position=1,
         desc='mask image')
-    extent = traits.Tuple((traits.Int, traits.Int, traits.Int),
+    extent = traits.Tuple(
+        (traits.Int, traits.Int, traits.Int),
         argstr='-extent %d,%d,%d',
         desc='set the window size of the denoising filter. (default = 5,5,5)')
     noise = File(
         argstr='-noise %s',
-        desc='the output noise map')
-    out_file = File(name_template='%s_denoised',
+        name_template='%s_noise',
         name_source='in_file',
         keep_extension=True,
+        desc='the output noise map',
+        genfile=True)
```

Comment:
Is `genfile` necessary? If I'm not mistaken, it's superseded by `name_template`/`name_source`.

Reply:
Yes, genfile should be removed here.